jupyter_server/services/sessions/sessionmanager.py

```diff
@@ -268,15 +276,34 @@ async def create_session(
         self._pending_sessions.remove(record)
         return result
 
-    def get_kernel_env(self, path):
-        """Return the environment variables that need to be set in the kernel"""
+    async def get_kernel_env(self, path: str, name: Optional[str] = None) -> Dict[str, str]:
+        """Return the environment variables that need to be set in the kernel
+
+        name: str
+            Here the name is likely to be the name of the associated file
+            with the current kernel at startup time.
+
+
+        """
+        if name is not None:
+            session_dir = await self.contents_manager.get_kernel_path(path=path)
+            cwd = self.kernel_manager.cwd_for_path(path)
+            path = os.path.join(cwd, session_dir, name)
+        assert isinstance(path, str)
```

Comment:
Won't `path` be the same value as `name` when the older form of the body is used (i.e., the one that includes a 'notebook' stanza)?  And, if so, will that produce the expected results?

Any idea how "old" the old API is?  I.e., does lab or nbclassic use that form of payload?

Reply:
I honestly don't know. It's honestly really unclear for me why this is still called `path` if it's often (always?) not a path.

I'm also quite confused about the on-disk Path and Api Path which is why I started to add type annotations.